Patch2

cyl3392207 opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [x] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

cyl3392207 committed:
Copy from 2016-04-01 version

cyl3392207 committed:
Adding parameters

parameters are added in policy defintion and assignment properties

cyl3392207 committed:
Adding description

azurecla commented:
Hi __@cyl3392207__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (mikech). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

anudeepsharma commented on arm-resources/policy/2016-12-01/swagger/policy.json:
```diff
@@ -0,0 +1,701 @@
+        "tags": [
+          "PolicyAssignments"
+        ],
+        "operationId": "PolicyAssignments_ListForResource",
```
Consider PolicyAssignment_ListByResource

cyl3392207 replied:
fixed

anudeepsharma commented on arm-resources/policy/2016-12-01/swagger/policy.json:
```diff
@@ -0,0 +1,701 @@
+        "x-ms-odata": "#/definitions/PolicyAssignment"
+      }
+    },
+    "/{policyAssignmentId}": {
```
rather than putting in description - /{scope}/providers/Microsoft.Authorization/policyAssignments/{policy-assignment-name}, I would prefer to have api as /{scope}/providers/Microsoft.Authorization/policyAssignments/{policy-assignment-name} delete, as why to put trouble to user to assemble this whole id himself he can use the api like PolicyAssignments.Delete(vm.Id, myPolicyName), with current swagger he has to do string manipulation like vm.Id + "/providers/Microsoft.Authorization/policyAssignments/" + myPolicyName.
I have not seen other RPs exposing DeleteById

cyl3392207 replied:
Using scope and name has been defined earlier.

anudeepsharma commented on arm-resources/policy/2016-12-01/swagger/policy.json:
```diff
@@ -0,0 +1,701 @@
+        "tags": [
+          "PolicyAssignments"
+        ],
+        "operationId": "PolicyAssignments_CreateById",
```
Again I would prefer taking 2 arguments scope and policyAssignmentName.

cyl3392207 replied:
Using scope and name has been defined earlier.

anudeepsharma commented on arm-resources/policy/2016-12-01/swagger/policy.json:
```diff
@@ -0,0 +1,701 @@
+        "tags": [
+          "PolicyAssignments"
+        ],
+        "operationId": "PolicyAssignments_GetById",
```
same as above 2 arguments rather than one argument

cyl3392207 replied:
Hey, Annaji. Can we review just he addition for this pull request? If you have comments on existing swagger spec, we can open a issue and address it in separate pull request. Does this sounds fair?

cyl3392207 replied:
Using scope and name has been defined earlier.

anudeepsharma commented on arm-resources/policy/2016-12-01/swagger/policy.json:
```diff
@@ -0,0 +1,701 @@
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The parent resource path.",
```
Not getting this description, please provide better description, that what this argument is, can it be empty, if yes then there will be so '/' in the path, will that not cause any problem.

cyl3392207 replied:
removed this as it doesn't seems correct and not necessary to have.

anudeepsharma commented on arm-resources/policy/2016-12-01/swagger/policy.json:
```diff
@@ -0,0 +1,697 @@
+        },
+        "parameters": {
+          "type": "object",
+          "description": "The policy parameters. Required if there is a parameter is used in policy rule. "
```
>Required if there is a parameter is used in policy rule. [](start = 49, length = 56)

Required if parameter is used in policy rule.

cyl3392207 replied:
fixed

anudeepsharma commented on arm-resources/policy/2016-12-01/swagger/policy.json:
```diff
@@ -0,0 +1,697 @@
+        },
+        "parameters": {
+          "type": "object",
+          "description": "The value of the policy parameters. Required if there is a parameter is used in policy rule. "
```
>Required if there is a parameter is used in policy rule.  [](start = 62, length = 57)

Required if a parameter is used in policy rule.

cyl3392207 replied:
fixed

anudeepsharma commented on arm-resources/policy/2016-12-01/swagger/policy.json:
```diff
@@ -0,0 +1,701 @@
+        },
+        "description":{
+          "type": "string",
+          "description": "The description of policy assignment. In case of policy violation, end user will see this message. "
```
>In case of policy violation, end user will see this message.  [](start = 64, length = 61)

This message will be part of response in case of policy violation.

Also remove blank spaces from the end.

cyl3392207 committed:
addressing comments

cyl3392207 committed:
reverting some comment

changing by to For in operation name
adding back old api and mark it as deprecated

AutorestCI commented:
No modification for Python

AutorestCI commented:
No modification for NodeJS

AutorestCI commented:
No modification for Ruby
